### PR TITLE
req. name clarification

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding prefixes for all requirements that are not meant to be package extras but rather developed

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2128.org.readthedocs.build/en/2128/

### CODE REVIEW

**Comment by Borda** [comment]
Interestingly seem you can't install extra which starts with `_` for example:
`pip install .[_typing]`

**Comment by SkafteNicki** [comment]
@Borda since we cannot prefix with `_` what are our options:
* Keep as it is?
* Move to a `requirements_dev` folder
* Prefix with `dev` (or something similar)

**Comment by Borda** [comment]
> since we cannot prefix with `_` what are our options

I think the other question is do we expect someone to have such extras as we ignored them already:
https://github.com/Lightning-AI/torchmetrics/blob/17b8460ef628a1578c69724884b45a9965937983/setup.py#L165

File: setup.py [hunk 1]
```diff
@@ -161,13 +161,20 @@ def _load_py_module(fname: str, pkg: str = "torchmetrics"):
...
 
-def _prepare_extras(
-    skip_files: Tuple[str] = ("base.txt", "devel.txt", "doctest.txt", "integrate.txt", "docs.txt")
-) -> dict:
+def _prepare_extras(skip_pattern: str = "^_", skip_files: Tuple[str] = ("base.txt",)) -> dict:
+    """Preparing extras for the package listing requirements.
+
+    Args:
+        skip_pattern: ignore files with this pattern, by default all files starting with _
+        skip_files: ignore some additional files, by default base requirements
```
Comment (Borda) [inline setup.py:169]: cc: @SkafteNicki

**Review by SkafteNicki — APPROVED** [review]
@Borda LGTM :)
I have added a small readme file just for newcomers to the project.

**Comment by Borda** [comment]
> I have added a small readme file just for newcomers to the project.

yes, that is great!

**Pull request merged**